ocs_ci/utility/pull_images_for_logging_42/pull_images.py

```diff
@@ -0,0 +1,79 @@
+from ocs_ci.utility.utils import run_cmd, clone_repo
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.exceptions import SyncImagesUnsuccessfulException
+
+import os
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_scripts_from_repo():
+    """
+    The function clones the repo quay-pull-images
+    """
+
+    url = 'https://github.com/red-hat-storage/quay-pull-images'
+    location = constants.QUAY_PULL_IMAGES_DIR
+    clone_repo(url, location, branch='master', to_checkout=None)
+
+
+def pull_packages_from_quay():
+    """
+    This is the function to pull the ocp 4.2 package manifests
+    for cluster-logging resources
+    """
+
+    registry = 'aosqe42'
+    version = "4.2"
+    path = f'{constants.QUAY_PULL_IMAGES_DIR}/pull_images_4.2'
+
+    get_scripts_from_repo()
+
+    # Fetch quay.token from ./data folder
+    quay_token_path = os.path.join(
+        constants.TOP_DIR,
+        "data",
+        "quay.token"
+    )
+    with open(quay_token_path, "r") as f:
+        quay_token = f.read()
+
+    # Download the repos and gets the images from repos and dumps it in
+    # OperatorSource_Images_Labels.txt to install
+
+    run_cmd(f"chmod +x {path}/getOperatorSourceMetadata.sh")
```

Comment:
for the compose the path please use os.path.join function. As I see the same path is used bellow, store it in variable and reuse on line 48 instead of composing the path again